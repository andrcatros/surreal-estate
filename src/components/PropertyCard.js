import React from "react";

const PropertyCard = ({ listing }) => {
  const { id, title, type, bathrooms, bedrooms, price, city, email } = listing;
  return (
    <div className="property-card" key={id}>
      <div className="listing-title">{title}</div>
      <div className="listing-type">{type}</div>
      <div className="listing-bathrooms">{bathrooms}</div>
      <div className="listing-bedrooms">{bedrooms}</div>
      <div className="listing-price">{price}</div>
      <div className="listing-city">{city}</div>
      <div className="listing-email">{email}</div>
    </div>
  );
};

export default PropertyCard;
