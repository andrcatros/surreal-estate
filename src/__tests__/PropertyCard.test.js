import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import PropertyCard from "../components/PropertyCard";

describe("renders PropertyCard component", () => {
  afterEach(cleanup);

  xit("renders placeholder text", () => {
    const { getByText } = render(<PropertyCard />);

    expect(getByText(/placeholder/i).textContent).toBe(
      "this is a placeholder the property card component"
    );
  });

  it("renders PropertyCard component with accurate data when passed props", () => {
    const mockPropertyListing = {
      id: "mockId",
      title: "mock listing",
      type: "flat",
      bedrooms: "3",
      price: 100000,
      city: "Manchester",
      email: "mock.m@email.com",
    };

    const { getByText, asFragment } = render(
      <PropertyCard listing={mockPropertyListing} />
    );

    expect(getByText("mock listing")).toHaveClass("listing-title");
    expect(getByText("flat")).toHaveClass("listing-type");
    expect(getByText("100000")).toHaveClass("listing-price");
    expect(getByText("Manchester")).toHaveClass("listing-city");
    expect(getByText("mock.m@email.com")).toHaveClass("listing-email");

    expect(asFragment()).toMatchSnapshot();
  });
});
