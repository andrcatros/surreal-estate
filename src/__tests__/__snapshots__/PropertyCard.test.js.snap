// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PropertyCard component renders PropertyCard component with accurate data when passed props 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <div
      class="listing-title"
    >
      mock listing
    </div>
    <div
      class="listing-type"
    >
      flat
    </div>
    <div
      class="listing-bathrooms"
    />
    <div
      class="listing-bedrooms"
    >
      3
    </div>
    <div
      class="listing-price"
    >
      100000
    </div>
    <div
      class="listing-city"
    >
      Manchester
    </div>
    <div
      class="listing-email"
    >
      mock.m@email.com
    </div>
  </div>
</DocumentFragment>
`;
